README.md

```diff
@@ -1 +1,165 @@
-# witnet-block-relay
\ No newline at end of file
+# witnet-ethereum-block-relay
+
+`witnet-ethereum-block-relay` is an open source implementation of Witnet Block Relay for EVM-compatible blockchains.
+
+
+ DISCLAIMER: this is a work in progress, by which we mean the contract could still be vulnerable to attack. Use at your own risk.
+
+## About WBI and the Block Relay
+
+The Bridge Nodes interact through the Witnet Bridge Interface (hereafter, "WBI"): a smart contract in charge of facilitating the communication between Witnet and smart contract platforms, such as Ethereum, click [here][wbi] for more information.
+The block headers in Witnet need to be available to the WBI, meaning that those headers need to be stored in some contract in Witnet, the Block Relay. In order to assure the validity of these block headers, the Block Relay must guarantee a consensus protocol and a Finality Gadget. The documentation about the stages of the Block Relay can be found [here][block-relay].
+
+## Connecting the Bridge and the Block Relay
+
+Due to the above, the more isolated the Block Relay and the WBI are, the more secure they are. Considering this, a proxy contract has been implemented, which connects to the WBI and from which the Block Relay is called. This allows to upgrade the Block Relay when necessary.
+The following diagram shows the flow between the WBI and the Block Relay.
+
+<p align=center>
+<img src="./images/wbi_br.jpg" width="550">
+</p>
+<p align=center>
+<em>Fig. 1: Connecting the WBI to the Block Relay.</em>
+</p>
+
+## BlockRelay based on ABS
+
+The `ABSBlockRelay` is a work in progress contract that implements the Active Bridge Set (ABS) as the set in charge of proposing and posting blocks to the Block Relay. The ABS is a subset of bridge nodes and is updated in the contract when posting a block through the WBI.
+
+A block header is later finalized when 2/3 of the ABS agree on a vote. More precisely, the flow is as follows:
+
+1. During each epoch, the members of the ABS (Active Bridge Set) propose votes for the previous epoch. The vote contains the previous vote that it extends.
+2. When the epoch is changed and another vote is proposed, if 2/3 of the members achieved consensus on a vote, the ´postNewBlock´ function is automatically called.
+3. When a vote is posted in the Block Relay, the previous votes that extends are finalized as well. If the previous vote is already finalized, it simply asserts that the vote that it extends is the correct one (i.e. the last block).
+
+The finality implemented in this contract is foundamental for the final purpose of a decentralized block relay. However the use of the ABS as the voting committee must be changed since these nodes are easily sybileable. Check the [Future steps](#future-steps) section to understand better the block relay design.
```

Comment:
```suggestion
The finality implemented in this contract is foundamental for the final purpose of a decentralized block relay. However the use of the ABS as the voting committee must be changed since these nodes are prone to Sybil attacks. Check the [Future steps](#future-steps) section to learn more about the block relay design.
```